microsetta_interface/templates/reports.jinja2

```diff
@@ -90,9 +90,23 @@
                 <div class="barcode-col barcode-text">
                     {{ sample.sample_barcode }}
                 </div>
-                <div class="barcode-col barcode-col-end" id="btn-view-{{ sample.sample_id }}">
-                    <span class="sample-label">{{ _('Not Received Yet') }}</span>
-                </div>
+                {% if sample.latest_scan_timestamp is none %}
+                    <div class="barcode-col barcode-col-end" id="btn-view-{{ sample.sample_id }}">
+                        <span class="sample-label">{{ _('Not Yet Received ') }} </span>
+                    </div>
+                {% elif sample.last_update is none %}
+                    <div class="barcode-col barcode-col-end" id="btn-view-{{ sample.sample_id }}">
+                        <span class="sample-label">{{ _('Sample Received - Information Needed ') }} </span>
```

Comment:
A sample which is received, and therefore has a scan date, could be perfect in its information in which case we do not need more information, right?

Reply:
The samples are already separated based on lock status, this loop is only through the samples that are not locked.